imbi/endpoints/reports/component_usage.py

```diff
@@ -0,0 +1,24 @@
+import re
+
+from tornado import web
+
+from imbi.endpoints import base
+
+
+class RequestHandler(base.RequestHandler):
+    NAME = 'reports-component-usage'
+    SQL = re.sub(
+        r'\s+', ' ', """\
+        SELECT c.name, c.package_url, c.active_version, c.status,
+               COUNT(v.id) AS version_count,
+               COUNT(p.project_id) AS project_count
+          FROM v1.components AS c
+          JOIN v1.component_versions AS v ON v.package_url = c.package_url
+          LEFT JOIN v1.project_components AS p ON p.version_id = v.id
+         GROUP BY c.name, c.package_url, c.active_version
+        """)
+
+    @web.authenticated
```

Comment:
Should the class instead inherit from `base.AuthenticatedRequestHandler`?

Reply:
I based this off of the other reports [which use web.authenticated](https://github.com/AWeber-Imbi/imbi-api/blob/66d2d7c35a125d8982cd4cdb33b3f63ef877d016/imbi/endpoints/reports/system_shs_history.py#L21) though they may have predated the stuff in `base`. It doesn't look like `base.AuthenticatedRequestHandler` adds any of the CRUD or OpenAPI goo. Lemme switch them and make sure that there aren't any surprises.

Reply:
Looks like that will work .... waiting on tests to pass locally before I push up the change.